pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -646,6 +654,82 @@ func (r *Reconciler) fetchRunningInstancesInInstanceGroup(projectID string, zone
 	return instanceSets, nil
 }
 
+// registerNewInstanceGroup registers an instance group when there is an instance
+// that is using that unkown instance group. After the creation of the instance group
+// we add patch our internal load balancer with that new instance group.
+func (r *Reconciler) registerNewInstanceGroup() error {
+	_, err := r.computeService.InstanceGroupInsert(r.projectID, r.providerSpec.Zone, &compute.InstanceGroup{
+		Name:       r.controlPlaneGroupName(),
+		Region:     r.providerSpec.Region,
+		Zone:       r.providerSpec.Zone,
+		Network:    r.instanceGroupNetworkName(),
+		Subnetwork: r.instanceGroupSubNetworkName(),
+	})
+	if err != nil {
+		return fmt.Errorf("instanceGroupInsert request failed: %v", err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) updateBackendServiceWithInstanceGroup() error {
+	// We have to wait for the instance group to be ready
+	backoff := wait.Backoff{
+		Duration: 10 * time.Second,
+		Steps:    10,
+		Factor:   1.0,
+		Jitter:   1.0,
+	}
+
+	errRetry := retry.OnError(backoff, func(error) bool { return true }, func() error {
+		backendServiceName := r.backendServiceName()
+
+		backendService, err := r.computeService.BackendServiceGet(r.projectID, r.providerSpec.Region, backendServiceName)
+		if err != nil {
+			return fmt.Errorf("backendServiceGet request failed: %v", err)
+		}
+
+		// Create backend that serves the backend service
+		backend := &compute.Backend{
+			BalancingMode: "CONNECTION",
+			Group:         r.FQDNInstanceGroup(),
+		}
+		backendService.Backends = append(backendService.Backends, backend)
+
+		_, err = r.computeService.AddInstanceGroupToBackendService(r.projectID, r.providerSpec.Region, backendServiceName, backendService)
+		if err != nil {
+			return fmt.Errorf("addInstanceGroupToBackendService request failed: %v", err)
+		}
+		return nil
+	})
+	if errRetry != nil {
+		return fmt.Errorf("retryOnConflict failed: %v", errRetry)
+	}
+
+	return nil
+}
+
+// FQDNInstanceGroup generates a FQDN for our instance group.
+// It is neccessary for the addition of the instance group to the backend service.
+func (r *Reconciler) FQDNInstanceGroup() string {
+	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s/instanceGroups/%s", r.projectID, r.providerSpec.Zone, r.controlPlaneGroupName())
+}
+
+// backendServiceName generates the name of a cluster's backend service
+func (r *Reconciler) backendServiceName() string {
+	return fmt.Sprintf("%s-api-internal", r.machine.Labels[machinev1.MachineClusterIDLabel])
+}
```

Comment:
Is the name of this always true? I wonder if in non-ipi installs this could vary, is there any way to look this up rather than assume it?

Reply:
This didn't cross my mind at all. In my installments, it was always in this format. I have no clue whether I can find that information somewhere. We can see that here as well: [https://console.cloud.google.com/compute/instanceGroups/list?project=openshift-gce-devel&pageState=(%22instanceGroupsTable%22:(%22f%22:%22%255B%255D%22))](GCP instanceGroupTable)
I guess the internal backend service, that get created by the installment is always in that format, but the user can have it's custom backend service?

Reply:
Lets leave as is for now and see how we get on, I think this should be ok as we only expect control plane machines to exist on clusters created by the installer